src/libsync/propagatedownload.cpp

```diff
@@ -457,13 +443,66 @@ void PropagateDownloadFile::startDownload()
         propagator()->_journal->commit("download file start");
     }
 
+    if (isZsyncPropagationEnabled(propagator(), _item)) {
+        if (_item->_previousSize) {
+            // Retrieve zsync metadata file from the server
+            qCInfo(lcZsyncGet) << "Retrieving zsync metadata file for:" << propagator()->getFilePath(_item->_file);
+            QNetworkRequest req;
+            req.setPriority(QNetworkRequest::LowPriority);
+
+            QUrl zsyncUrl = Utility::concatUrlPath(propagator()->account()->davUrl(),
+                                                   propagator()->_remoteFolder + _item->_file,
+                                                   QList<QPair<QString, QString>> ({{"zsync", nullptr}}));
+            auto job = propagator()->account()->sendRequest("GET", zsyncUrl, req);
+            connect(job, &SimpleNetworkJob::finishedSignal, this, &PropagateDownloadFile::slotZsyncGetMetaFinished);
+            return;
+        }
+        qCInfo(lcZsyncGet) << "No local copy of:" << propagator()->getFilePath(_item->_file);
+    }
+
+    startDownloadNormal();
+}
+
+
+void PropagateDownloadFile::slotZsyncGetMetaFinished(QNetworkReply *reply)
+{
+    int httpStatusCode = reply->attribute(QNetworkRequest::HttpStatusCodeAttribute).toInt();
+    if (httpStatusCode != 200) {
+        qCInfo(lcPropagateDownload)
+            << "No remote zsync file for:"
+            << propagator()->getFilePath(_item->_file);
+        startDownloadNormal();
+        return;
+    }
+
+    QByteArray zsyncData = reply->readAll();
+
+    qCInfo(lcZsyncGet) << "Got remote zsync file for:" << _item->_file << "size:" << zsyncData.size();
+
+    QMap<QByteArray, QByteArray> headers;
+    _job = new GETFileZsyncJob(propagator()->account(), _item, propagator(),
+                               propagator()->_remoteFolder + _item->_file,
+                               propagator()->getFilePath(_item->_file),
```

Comment:
Can't this two values be calculated in the `GETFileZsyncJob()` constructor rather than cluttering the parameter list that much?

Reply:
To be honest about the @guruz approach, it's definitely nicer, but I'm not spending another two weeks on this PR for $1k. If you guys don't want to merge, then so be it.

About the params for `GETFileZsyncJob`, yes this is left over from old implementations, will fix.

Reply:
Done.